hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/SnapshotChainManager.java

```diff
@@ -499,23 +468,24 @@ public String nextPathSnapshot(String snapshotPath, String snapshotID)
    * @return String, snapshot UUID of previous snapshot in chain from
    * snapshotID
    */
-  public String previousPathSnapshot(String snapshotPath, String snapshotID)
-      throws NoSuchElementException {
+  public String previousPathSnapshot(String snapshotPath, String snapshotID) {
     if (!snapshotChainPath.containsKey(snapshotPath) ||
         !snapshotChainPath.get(snapshotPath).containsKey(snapshotID)) {
-      LOG.error("no snapshot for provided snapshotPath {} and "
-          + " snapshotID {}", snapshotPath, snapshotID);
-      throw new NoSuchElementException("No snapshot: " + snapshotID
-          + " for snapshot path:" + snapshotPath);
+      LOG.error("No snapshot found for provided snapshotId: {} and " +
+          "snapshotPath: {}", snapshotID, snapshotPath);
+      throw new NoSuchElementException(String.format("SnapshotId: %s is not " +
+              "found in snapshot chain for snapshotPath: %s.", snapshotID,
+          snapshotPath));
     }
     if (snapshotChainPath
         .get(snapshotPath)
         .get(snapshotID)
         .getPreviousSnapshotID() == null) {
-      LOG.error("no preceeding snapshot for provided snapshotPath {}, "
-          + "snapshotID {}", snapshotPath, snapshotID);
-      throw new NoSuchElementException("no preceeding snapshot from: "
-          + snapshotID + "for snapshot path:" + snapshotPath);
+      LOG.error("No preceding snapshot for provided snapshotId: {} and " +
+          "snapshotPath: {}", snapshotPath, snapshotID);
+      throw new NoSuchElementException(String.format("Ne preceding snapshot " +
```

Comment:
nit:
```suggestion
      throw new NoSuchElementException(String.format("No preceding snapshot " +
```

Reply:
Thanks for detailed review @aswinshakil. Updated as suggested.